Networking/Networking/Remote/JetpackConnectionRemote.swift

```diff
@@ -19,16 +19,26 @@ public final class JetpackConnectionRemote: Remote {
 
         enqueue(request, mapper: mapper, completion: completion)
     }
+
+    /// Fetches the user connection state with the site's Jetpack.
+    ///
+    public func fetchJetpackUser(completion: @escaping (Result<JetpackUser, Error>) -> Void) {
+        let request = WordPressOrgRequest(baseURL: siteURL, method: .get, path: Path.jetpackConnectionData)
+        let mapper = JetpackUserMapper()
+        enqueue(request, mapper: mapper, completion: completion)
+    }
 }
 
 public extension JetpackConnectionRemote {
     enum ConnectionError: Int, Error {
         case malformedURL
+        case currentUserNotFound
     }
 }
 
 private extension JetpackConnectionRemote {
     enum Path {
         static let jetpackConnectionURL = "/jetpack/v4/connection/url"
+        static let jetpackConnectionData = "/jetpack/v4/connection/data"
```

Comment:
super nit: how about mentioning it's to fetch the Jetpack user given its usage in `fetchJetpackUser`?
```suggestion
        static let jetpackConnectionUser = "/jetpack/v4/connection/data"
```